code/game/objects/structures/mirror.dm

```diff
@@ -37,29 +37,51 @@
 					return
 		var/userloc = H.loc
 
+		
+		var/which = alert("What would you like to change?", "Appearance", "Hair", "Beard", "Undies")
+
+		if((!which) || (userloc != H.loc))
```

Comment:
an adjacency check would be preferred I think, as long as you remain in slapping range of that mirror

Reply:
wilco